golem-cli/src/main.rs

```diff
@@ -14,20 +14,29 @@
 
 extern crate derive_more;
 
-use clap::Parser;
+use clap::{CommandFactory, Parser};
 use clap_verbosity_flag::{Level, Verbosity};
 use golem_cli::command::profile::OssProfileAdd;
 use golem_cli::config::{Config, NamedProfile, Profile};
+use golem_cli::init::{CliKind, DummyProfileAuth, GolemInitCommand};
+use golem_cli::oss;
+use golem_cli::oss::command::{GolemOssCommand, OssCommand};
 use indoc::eprintdoc;
 use std::path::PathBuf;
 use tracing::info;
 use tracing_subscriber::FmtSubscriber;
 
-use golem_cli::init::{CliKind, DummyProfileAuth, GolemInitCommand};
-use golem_cli::oss;
-use golem_cli::oss::command::GolemOssCommand;
-
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let command = GolemOssCommand::<OssProfileAdd>::parse();
```

Comment:
This breaks `--help` on not initialised CLI.
Please do not add new command parse actions.

Reply:
Do not change `src/main.rs` add handlers to `async_main` in `src/oss/main.rs` and `src/init.rs`

Reply:
it does not look like it will break help
```
golem % ./target/debug/golem-cli worker connect --component-name cart2 --worker-name aaa
Error: Your Golem CLI is not configured. Please run `golem-cli init`
golem % ./target/debug/golem-cli --help
Command line interface for OSS version of Golem.

Usage: golem-cli [OPTIONS] <COMMAND>

Commands:
  component       Upload and manage Golem components
  worker          Manage Golem workers
  get             Get resource by URI
  new             Create a new Golem component from built-in examples
  list-examples   Lists the built-in examples available for creating new components
  stubgen         WASM RPC stub generator
  api-definition  Manage Golem api definitions
  api-deployment  Manage Golem api deployments
  profile         Manage profiles
  init            Interactively creates default profile
  completion      Generate shell completions
  help            Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose...
          Increase logging verbosity

  -q, --quiet...
          Decrease logging verbosity

  -F, --format <FORMAT>


  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version
golem % ./target/debug/golem-cli worker --help
Manage Golem workers

Usage: golem-cli worker [OPTIONS] <COMMAND>

Commands:
  add               Creates a new idle worker
  idempotency-key   Generates an idempotency key for achieving at-most-one invocation when doing retries
  invoke-and-await  Invokes a worker and waits for its completion
  invoke            Triggers a function invocation on a worker without waiting for its completion
  connect           Connect to a worker and live stream its standard output, error and log channels
  interrupt         Interrupts a running worker
  resume            Resume an interrupted worker
  simulated-crash   Simulates a crash on a worker for testing purposes
  delete            Deletes a worker
  get               Retrieves metadata about an existing worker
  list              Retrieves metadata about an existing workers in a component
  update            Updates a worker
  help              Print this message or the help of the given subcommand(s)

Options:
  -v, --verbose...       Increase logging verbosity
  -q, --quiet...         Decrease logging verbosity
  -F, --format <FORMAT>
  -h, --help             Print help

```

Reply:
It is not the correct command line.

Reply:
Try `golem-cli --help` - there will be no `Your Golem CLI is not configured.`